posts/2021-03-24-whats-new-in-MP-Rest-Client2.0.adoc

```diff
@@ -0,0 +1,212 @@
+                                   .property("com.ibm.ws.jaxrs.client.proxy.password", "12345") //same as my luggage! :)
+                                   //...
+                                   .build(MyClient.class);
+----
+
+[#auto-redirects]
+== Automatically following redirects
+
+If a RESTful resource has been relocated, often the HTTP response code will be in the 300 range and will indicate the
+new location. Rather than handling the 3XX response and manually issuing a new request, MP Rest Client 2.0 allows rest
+client instances to automatically follow redirects. This can be configured programmatically while building the client
+instance or via MP Config. Here is an example of configuring auto-redirect via the `RestClientBuilder` API:
+
+[source,java]
+----
+MyClient client = RestClientBuilder.newBuilder()
+                                   .followRedirects(true)
+                                   //...
+                                   .build(MyClient.class);
+----
+
+And here is how you would configure it via MP Config:
+
+`com.mypkg.MyClient/mp-rest/followRedirects=true`
+
+[#sse-support]
+== Support for Server Sent Events (SSEs)
+
+link:https://openliberty.io/guides/reactive-messaging-sse.html[Server Sent Events] are part of the HTML 5 spec and
+enables a server to push data to a client asynchronously via events - all over HTTP. The JAX-RS 2.1 spec enabled SSE
+support for both the client and server. Now you can consume SSE events from the type-safe MP Rest Client.
+
+The MP Rest Client specification uses the link:http://www.reactive-streams.org/[Reactive Streams] APIs to consume
+events. A client interface capable of consuming SSEs would look something like this:
+
+[source,java]
+----
+@RegisterRestClient
+public interface SseClient {
+
+    @GET
+    @Path("/path/sse")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    Publisher<String> getStrings();
+
+    @GET
+    @Path("/path/sse2")
+    @Produces(MediaType.SERVER_SENT_EVENTS)
+    Publisher<InboundSseEvent> getEvents();
+}
+----
+
+First, the method (or interface) must be annotated with `@Produces(MediaType.SERVER_SENT_EVENTS)` to indicate that it
+expects the server to produce SSEs. Next, the method's return type must be `org.reactivestreams.Publisher`. The generic
+type can be `javax.ws.rs.sse.InboundSseEvent` (from JAX-RS), a primitive, a String or a complex type. Complex types may only be used if;
+
+1) the server only sends one type of event (e.g. only sends `WeatherEvents` - then `Publisher<WeatherEvent>` would be applicable)
+
+_and_
+
+2) there is a registered entity provider capable of converting the events (e.g. `MessageBodyReader<WeatherEvent>`).
+
+In most cases, if the remote server sends events using JSON, you can enable the `jsonb-1.0` feature in your Liberty
+server which will automatically register a JSON-B-based entity provider.
+
+
+Once you've invoked one of these methods, you should be able to register one or more `Subscriber` instances to the
```

Comment:
avoid past perfect and subjunctive:

Once you invoke one of these methods, you can register one or more `Subscriber` instances...